[BUG] Java Doc warnings with gradle check

### What is the bug?
Gradle check throws a bunch of java doc warnings. 

### How can one reproduce the bug?
- Checkout SDK repo
- Run `./gradlew check`

```
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/ActionExtension.java:48: warning: no @return
    default List<ActionHandler<? extends ActionRequest, ? extends ActionResponse>> getActions() {
                                                                                   ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/ActionExtension.java:56: warning: no @return
    default List<ActionType<? extends ActionResponse>> getClientActions() {
                                                       ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/ActionExtension.java:63: warning: no @return
    default List<ActionFilter> getActionFilters() {
                               ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/ActionExtension.java:79: warning: no @return
    default Collection<RestHeaderDefinition> getRestHeaders() {
                                             ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/ActionExtension.java:86: warning: no @return
    default Collection<String> getTaskHeaders() {
                               ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/ActionExtension.java:109: warning: no @param for threadContext
    default UnaryOperator<ExtensionRestHandler> getRestHandlerWrapper(ThreadContext threadContext) {
                                                ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/ActionExtension.java:109: warning: no @return
    default UnaryOperator<ExtensionRestHandler> getRestHandlerWrapper(ThreadContext threadContext) {
                                                ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/ActionExtension.java:177: warning: no @return
    default Collection<RequestValidators.RequestValidator<PutMappingRequest>> mappingRequestValidators() {
                                                                              ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/ActionExtension.java:185: warning: no @return
    default Collection<RequestValidators.RequestValidator<IndicesAliasesRequest>> indicesAliasesRequestValidators() {
                                                                                  ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/BaseExtensionRestHandler.java:103: warning: no @param for e
    protected ExtensionRestResponse exceptionalRequest(ExtensionRestRequest request, Exception e) {
                                    ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/ActionExtension.java:125: warning: no @param for action
        public ActionHandler(
               ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/ActionExtension.java:125: warning: no @param for transportAction
        public ActionHandler(
               ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/ActionExtension.java:125: warning: no @param for supportTransportActions
        public ActionHandler(
               ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/EngineExtension.java:36: warning: no @param for indexSettings
    default Optional<EngineFactory> getEngineFactory(IndexSettings indexSettings) {
                                    ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/EngineExtension.java:47: warning: no @param for indexSettings
    default Optional<CodecServiceFactory> getCustomCodecServiceFactory(IndexSettings indexSettings) {
                                          ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/EngineExtension.java:47: warning: no @return
    default Optional<CodecServiceFactory> getCustomCodecServiceFactory(IndexSettings indexSettings) {
                                          ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SDKClient.java:288: warning: no @return
        public SDKRestClient admin() {
                             ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SDKClient.java:295: warning: no @return
        public SDKClusterAdminClient cluster() {
                                     ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SDKClient.java:319: warning: no @return
        public SDKIndicesClient indices() {
                                ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:74: warning: no @return
    default List<ScoreFunctionSpec<?>> getScoreFunctions() {
                                       ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:82: warning: no @return
    default List<SignificanceHeuristicSpec<?>> getSignificanceHeuristics() {
                                               ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:90: warning: no @return
    default List<SearchExtensionSpec<MovAvgModel, MovAvgModel.AbstractModelParser>> getMovingAverageModels() {
                                                                                    ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:97: warning: no @param for context
    default List<FetchSubPhase> getFetchSubPhases(FetchPhaseConstructionContext context) {
                                ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:97: warning: no @return
    default List<FetchSubPhase> getFetchSubPhases(FetchPhaseConstructionContext context) {
                                ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:104: warning: no @return
    default List<SearchExtSpec<?>> getSearchExts() {
                                   ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:111: warning: no @return
    default Map<String, Highlighter> getHighlighters() {
                                     ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:118: warning: no @return
    default List<SuggesterSpec<?>> getSuggesters() {
                                   ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:125: warning: no @return
    default List<QuerySpec<?>> getQueries() {
                               ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:132: warning: no @return
    default List<SortSpec<?>> getSorts() {
                              ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:139: warning: no @return
    default List<AggregationSpec> getAggregations() {
                                  ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:164: warning: no @return
    default List<PipelineAggregationSpec> getPipelineAggregations() {
                                          ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:171: warning: no @return
    default List<RescorerSpec<?>> getRescorers() {
                                  ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:180: warning: no @return
    default Optional<QueryPhaseSearcher> getQueryPhaseSearcher() {
                                         ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:189: warning: no @return
    default Optional<ExecutorServiceProvider> getIndexSearcherExecutorProvider() {
                                              ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:821: warning: no @return
        public ParseField getName() {
                          ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:828: warning: no @return
        public Writeable.Reader<? extends W> getReader() {
                                             ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:835: warning: no @return
        public P getParser() {
                 ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:425: warning: no @param for <T>
        public <T extends AggregationBuilder> AggregationSpec(
                                              ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:443: warning: no @param for <T>
        public <T extends AggregationBuilder> AggregationSpec(String name, Writeable.Reader<T> reader, ContextParser<String, T> parser) {
                                              ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:484: warning: no @param for resultReader
        public AggregationSpec addResultReader(Writeable.Reader<? extends InternalAggregation> resultReader) {
                               ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:484: warning: no @return
        public AggregationSpec addResultReader(Writeable.Reader<? extends InternalAggregation> resultReader) {
                               ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:491: warning: no @param for writeableName
        public AggregationSpec addResultReader(String writeableName, Writeable.Reader<? extends InternalAggregation> resultReader) {
                               ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:491: warning: no @param for resultReader
        public AggregationSpec addResultReader(String writeableName, Writeable.Reader<? extends InternalAggregation> resultReader) {
                               ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:491: warning: no @return
        public AggregationSpec addResultReader(String writeableName, Writeable.Reader<? extends InternalAggregation> resultReader) {
                               ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:499: warning: no @return
        public Map<String, Writeable.Reader<? extends InternalAggregation>> getResultReaders() {
                                                                            ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:507: warning: no @return
        public Consumer<ValuesSourceRegistry.Builder> getAggregatorRegistrar() {
                                                      ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:515: warning: no @param for aggregatorRegistrar
        public AggregationSpec setAggregatorRegistrar(Consumer<ValuesSourceRegistry.Builder> aggregatorRegistrar) {
                               ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:515: warning: no @return
        public AggregationSpec setAggregatorRegistrar(Consumer<ValuesSourceRegistry.Builder> aggregatorRegistrar) {
                               ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:848: warning: no description for @param
         * @param highlighters
           ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:671: warning: no @param for parser
        public PipelineAggregationSpec(
               ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:683: warning: no @param for resultReader
        public PipelineAggregationSpec addResultReader(Writeable.Reader<? extends InternalAggregation> resultReader) {
                                       ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:683: warning: no @return
        public PipelineAggregationSpec addResultReader(Writeable.Reader<? extends InternalAggregation> resultReader) {
                                       ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:690: warning: no @param for writeableName
        public PipelineAggregationSpec addResultReader(String writeableName, Writeable.Reader<? extends InternalAggregation> resultReader) {
                                       ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:690: warning: no @param for resultReader
        public PipelineAggregationSpec addResultReader(String writeableName, Writeable.Reader<? extends InternalAggregation> resultReader) {
                                       ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:690: warning: no @return
        public PipelineAggregationSpec addResultReader(String writeableName, Writeable.Reader<? extends InternalAggregation> resultReader) {
                                       ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:698: warning: no @return
        public Map<String, Writeable.Reader<? extends InternalAggregation>> getResultReaders() {
                                                                            ^
/home/ubuntu/opensearch-sdk-java/src/main/java/org/opensearch/sdk/SearchExtension.java:334: warning: no @return
        public Writeable.Reader<? extends Suggest.Suggestion> getSuggestionReader() {
                                                              ^
57 warnings
```
### What is the expected behavior?
No errors :)

### What is your proposed fix?
There are 2 parts to this issue:
- Force gradle check to fail when java docs has warnings
- Fix all java docs warnings

--- a/build.gradle
+++ b/build.gradle
@@ -17,6 +17,8 @@ plugins {
     id "com.diffplug.spotless" version "6.16.0" apply false
     id 'jacoco'
     id "com.form.diff-coverage" version "0.9.5"
+    // for javadocs and checks spotless doesn't do
+    id 'checkstyle'
 }
 
 
@@ -71,10 +73,6 @@ repositories {
     maven { url "https://d1nvenhzbhpy0q.cloudfront.net/snapshots/lucene/"}
 }
 
-configurations {
-  requireJavadoc
-}
-
 dependencies {
 
     def opensearchVersion = "3.0.0-SNAPSHOT"
@@ -87,7 +85,6 @@ dependencies {
     def junit5Version = "5.9.2"
     def junitPlatform = "1.9.2"
     def jaxbVersion = "2.3.1"
-    def requireJavadocVersion = "1.0.6"
 
     implementation("org.opensearch:opensearch:${opensearchVersion}")
     implementation("org.apache.logging.log4j:log4j-api:${log4jVersion}")
@@ -117,33 +114,7 @@ dependencies {
     testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:${junit5Version}")
     testImplementation("org.opensearch.test:framework:${opensearchVersion}")
     testRuntimeOnly("org.junit.platform:junit-platform-launcher:${junitPlatform}")
-    requireJavadoc("org.plumelib:require-javadoc:${requireJavadocVersion}")
 }
-
-// this task tests for the presence of javadocs but not the content/tags
-task requireJavadoc(type: JavaExec) {
-  group = 'Documentation'
-  description = 'Ensures that Javadoc documentation exists.'
-  mainClass = "org.plumelib.javadoc.RequireJavadoc"
-  classpath = configurations.requireJavadoc
-  args "src/main/java"
-  // javadocs on private methods optional
-  args "--dont-require-private=true"
-  // javadocs on trivial getters/setters optional
-  args "--dont-require-trivial-properties"
-}
-check.dependsOn requireJavadoc
-
-// this task checks the content/tags of existing javadocs
-task javadocStrict(type: Javadoc) {
-  group = 'Documentation'
-  description = 'Run Javadoc in strict mode: with -Xdoclint:all, on all members.'
-  source = sourceSets.main.allJava
-  classpath = sourceSets.main.runtimeClasspath
-  options.addStringOption('Xdoclint:all', '-quiet')
-  options.memberLevel = JavadocMemberLevel.PRIVATE
-}
-check.dependsOn javadocStrict
 
 // this task runs the helloworld sample extension
 task helloWorld(type: JavaExec) {

--- a/config/checkstyle/checkstyle.xml
+++ b/config/checkstyle/checkstyle.xml
@@ -1,0 +1,54 @@
+<?xml version="1.0"?>
+<!--
+  ~ SPDX-License-Identifier: Apache-2.0
+
+  ~ The OpenSearch Contributors require contributions made to
+  ~ this file be licensed under the Apache-2.0 license or a
+  ~ compatible open source license.
+  -->
+
+<!DOCTYPE module PUBLIC
+        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
+        "https://checkstyle.org/dtds/configuration_1_3.dtd">
+
+<!--
+    Formatting checks are done by spotless. This config is only for checks spotless doesn't do.
+-->
+
+<module name="Checker">
+    <property name="charset" value="UTF-8" />
+
+    <module name="SuppressionFilter">
+        <property name="file" value="${config_loc}/checkstyle_suppressions.xml" />
+    </module>
+
+    <module name="TreeWalker">
+        <!-- Disallows star imports -->
+        <module name="AvoidStarImport" />
+        
+        <!--  Requires javadoc on public and protected interfaces, classes, enums -->
+        <module name="MissingJavadocType">
+            <property name="scope" value="protected"/>
+        </module>
+        
+        <!-- Requires javadoc on public and protected methods, excluding getters and setters -->
+        <module name="MissingJavadocMethod">
+            <property name="scope" value="protected"/>
+            <property name="allowMissingPropertyJavadoc" value="true"/>
+        </module>
+        
+        <!--  Checks that javadocs contain correct param and return -->
+        <module name="JavadocMethod">
+            <property name="id" value="JavadocMethod"/>
+            <property name="accessModifiers" value="public, protected"/>
+        </module>
+
+        <!-- Allows missing returns on extension interface methods -->
+        <module name="JavadocMethod">
+            <property name="id" value="JavadocMethodAllowMissingReturnTag"/>
+            <property name="accessModifiers" value="public, protected"/>
+            <property name="allowMissingReturnTag" value="true"/>
+        </module>
+        
+    </module>
+</module>

--- a/config/checkstyle/checkstyle_suppressions.xml
+++ b/config/checkstyle/checkstyle_suppressions.xml
@@ -1,0 +1,20 @@
+<?xml version="1.0"?>
+<!--
+  ~ SPDX-License-Identifier: Apache-2.0
+
+  ~ The OpenSearch Contributors require contributions made to
+  ~ this file be licensed under the Apache-2.0 license or a
+  ~ compatible open source license.
+  -->
+
+<!DOCTYPE suppressions PUBLIC
+        "-//Checkstyle//DTD SuppressionFilter Configuration 1.1//EN"
+        "https://checkstyle.org/dtds/suppressions_1_1.dtd">
+
+<suppressions>
+    <!-- No javadoc checks on test classes -->
+    <suppress checks="Javadoc*" files="[\\/]src[\\/]test[\\/].*" />
+    <!-- No return required for Extension interface classes -->
+    <suppress id="JavadocMethod" files="\w+Extension\.java" />
+    <suppress id="JavadocMethodAllowMissingReturnTag" files="^((?!\w+Extension\.java).)*$" />
+</suppressions>

--- a/gradle/formatting.gradle
+++ b/gradle/formatting.gradle
@@ -11,13 +11,6 @@ allprojects {
             trimTrailingWhitespace()
             endWithNewline();
 
-            custom 'Refuse wildcard imports', {
-                // Wildcard imports can't be resolved; fail the build
-                if (it =~ /\s+import .*\*;/) {
-                    throw new AssertionError("Do not use wildcard imports.  'spotlessApply' cannot resolve this issue.")
-                }
-            }
-
             // See DEVELOPER_GUIDE.md for details of when to enable this.
             if (System.getProperty('spotless.paddedcell') != null) {
                 paddedCell()

--- a/src/main/java/org/opensearch/sdk/ActionExtension.java
+++ b/src/main/java/org/opensearch/sdk/ActionExtension.java
@@ -105,6 +105,8 @@ public interface ActionExtension {
      *    }
      * }
      * </pre>
+     *
+     * @param threadContext The Thread Context which can be used by the operator
      */
     default UnaryOperator<ExtensionRestHandler> getRestHandlerWrapper(ThreadContext threadContext) {
         return null;

--- a/src/main/java/org/opensearch/sdk/BaseExtensionRestHandler.java
+++ b/src/main/java/org/opensearch/sdk/BaseExtensionRestHandler.java
@@ -98,6 +98,7 @@ public abstract class BaseExtensionRestHandler implements ExtensionRestHandler {
      * Returns a default response when a request handler throws an exception.
      *
      * @param request The request that caused the exception
+     * @param e The exception
      * @return an ExtensionRestResponse identifying the exception
      */
     protected ExtensionRestResponse exceptionalRequest(ExtensionRestRequest request, Exception e) {

--- a/src/main/java/org/opensearch/sdk/EngineExtension.java
+++ b/src/main/java/org/opensearch/sdk/EngineExtension.java
@@ -31,6 +31,7 @@ public interface EngineExtension {
      * {@link Optional#empty()}. If multiple Extensions return an engine factory for a given index the index will not be created and an
      * {@link IllegalStateException} will be thrown during index creation.
      *
+     * @param indexSettings the index settings to inspect
      * @return an optional engine factory
      */
     default Optional<EngineFactory> getEngineFactory(IndexSettings indexSettings) {
@@ -43,6 +44,9 @@ public interface EngineExtension {
      * to determine if a custom {@link CodecServiceFactory} should be provided for the given index. A Extension that is not overriding
      * the {@link CodecServiceFactory} through the Extension can ignore this method and the default Codec specified in the
      * {@link IndexSettings} will be used.
+     *
+     * @param indexSettings the index settings to inspect
+     * @return an optional engine factory
      */
     default Optional<CodecServiceFactory> getCustomCodecServiceFactory(IndexSettings indexSettings) {
         return Optional.empty();

--- a/src/main/java/org/opensearch/sdk/SDKClient.java
+++ b/src/main/java/org/opensearch/sdk/SDKClient.java
@@ -294,6 +294,8 @@ public class SDKClient implements Closeable {
 
         /**
          * The admin client that can be used to perform administrative operations.
+         *
+         * @return An instance of this client. Method provided for backwards compatibility.
          */
         public SDKRestClient admin() {
             return this;
@@ -301,6 +303,8 @@ public class SDKClient implements Closeable {
 
         /**
          * A client allowing to perform actions/operations against the cluster.
+         *
+         * @return An instance of a cluster admin client.
          */
         public SDKClusterAdminClient cluster() {
             return new SDKClusterAdminClient(restHighLevelClient.cluster());
@@ -325,6 +329,8 @@ public class SDKClient implements Closeable {
 
         /**
          * A client allowing to perform actions/operations against the indices.
+         *
+         * @return An instance of an indices client.
          */
         public SDKIndicesClient indices() {
             return new SDKIndicesClient(restHighLevelClient.indices());

--- a/src/main/java/org/opensearch/sdk/SearchExtension.java
+++ b/src/main/java/org/opensearch/sdk/SearchExtension.java
@@ -93,6 +93,8 @@ public interface SearchExtension {
 
     /**
      * The new {@link FetchSubPhase}s defined by this Extension.
+     *
+     * @param context The context for which to fetch the subphases.
      */
     default List<FetchSubPhase> getFetchSubPhases(FetchPhaseConstructionContext context) {
         return emptyList();

